src/rlang/df.c

```diff
@@ -0,0 +1,70 @@
+#include "rlang.h"
+
+sexp* r_classes_data_frame = NULL;
+sexp* r_classes_tibble = NULL;
+
+#include "decl/df-decl.h"
+
+
+sexp* r_alloc_data_frame(r_ssize n_rows,
+                         sexp* names,
+                         enum r_type* v_types, r_ssize types_size) {
+  sexp* df = KEEP(r_new_list(types_size));
+
+  if (r_typeof(names) != r_type_character) {
+    r_abort("`names` must be a character vector.");
```

Comment:
Not sure if you care about `NA_character_` names here or not

Reply:
hmm since the names are provided by a developer rather than a user I'd prefer to avoid doing extra checks.